monix-catnap/shared/src/main/scala/monix/catnap/ConcurrentChannel.scala

```diff
@@ -924,6 +924,10 @@ object ConcurrentChannel {
           }
       }
     }
+
+    override def isEmpty: F[Boolean] =
+      F.pure(queue.isEmpty)
```

Comment:
Should be `F.delay`. Testing the state of a mutable queue is not referentially transparent.